packages/bundler/src/modules/ExecutionManager.ts

```diff
@@ -35,16 +36,17 @@ export class ExecutionManager {
    * @param userOp the UserOp to send.
    * @param entryPointInput the entryPoint passed through the RPC request.
    */
-  async sendUserOperation (userOp: UserOperation, entryPointInput: string): Promise<void> {
+  async sendUserOperation (userOp: OperationBase, entryPointInput: string): Promise<void> {
     await this.mutex.runExclusive(async () => {
       debug('sendUserOperation')
       this.validationManager.validateInputParameters(userOp, entryPointInput)
       const validationResult = await this.validationManager.validateUserOp(userOp, undefined)
-      const userOpHash = await this.validationManager.entryPoint.getUserOpHash(packUserOp(userOp))
+      const userOpHash = await this.validationManager.getOperationHash(userOp)
       await this.depositManager.checkPaymasterDeposit(userOp)
       this.mempoolManager.addUserOp(userOp,
         userOpHash,
-        validationResult.returnInfo.prefund,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
I think "?? 0" better than "!"

Reply:
Done. We should make more refactoring here in terms of UserOps/Type4Tx differences, but that can be done later.